test/e2e/common/pods.go

```diff
@@ -433,15 +434,19 @@ var _ = framework.KubeDescribe("Pods", func() {
 			},
 		}
 
-		f.TestContainerOutput("service env", pod, 0, []string{
+		const maxRetries = 3
+		expectedVars := []string{
 			"FOOSERVICE_SERVICE_HOST=",
 			"FOOSERVICE_SERVICE_PORT=",
 			"FOOSERVICE_PORT=",
 			"FOOSERVICE_PORT_8765_TCP_PORT=",
 			"FOOSERVICE_PORT_8765_TCP_PROTO=",
 			"FOOSERVICE_PORT_8765_TCP=",
 			"FOOSERVICE_PORT_8765_TCP_ADDR=",
-		})
+		}
+		framework.ExpectNoErrorWithRetries(func() error {
```

Comment:
This doesn't work, does it?

This gets the logs N times, but the logs will always be the same because it's a `RestartNever` container.

We need to actually *run* the container multiple distinct times in order for environment changes to take effect.